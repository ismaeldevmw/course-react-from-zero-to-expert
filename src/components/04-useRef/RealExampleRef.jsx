import React from 'react';
import { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';
import './useref.css';

export const RealExampleRef = () => {
    
    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr/>

            { show && <MultipleCustomHooks /> }

            <button 
                className="btn btn-primary"
                onClick={() => {
                    setShow(!show);  
                }}
            >
                Show / Hide
            </button>
        </div>
    )
}
