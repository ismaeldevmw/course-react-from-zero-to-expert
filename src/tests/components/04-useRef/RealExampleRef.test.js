import { shallow } from 'enzyme';
import React from 'react';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Tests in <RealExampleRef />', () => {
    

    const wrapper = shallow( <RealExampleRef />)

    test('should display correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);

    });

    test('should show the component <MultipleCustomHooks />', () => {

        wrapper.find('button').simulate('click');

        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);

    });

});