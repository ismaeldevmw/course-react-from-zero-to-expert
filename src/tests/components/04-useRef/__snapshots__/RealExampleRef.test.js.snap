// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RealExampleRef /> should display correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
